### PR TITLE
Support zuul nodepool disable systemd unit

### PR DESCRIPTION
**Pull request by liusheng**
This change clean some redundant variable definition and disable
nodepool and zuul services systemd units to make these services do not
started when system booted.

Additionally, the current testing roles only test the service status, it is meaningless, so just removed them.

### CODE REVIEW

File: inventory/test-allinone-ha.yaml [hunk 1]
```diff
@@ -44,47 +44,45 @@ all:
...
-                service_state: stopped
+                zuul_service_zuul_fingergw_state: stopped
         zuul-log-server:
           children:
             logserver-master:
               hosts: allinone01
             logserver-slave:
               hosts: allinone02
-              vars:
-                service_state: stopped
```
Comment (huangtianhua) [inline inventory/test-allinone-ha.yaml:87]: Do we have to change the var to logserver_state instead delete the var directly?

File: inventory/allinone-ha.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@ all:
             nodepool-builder-slave:
               hosts: allinone02
               vars:
-                service_state: stopped
+                nodepool_service_nodepool_builder_state: stopped
```
Comment (huangtianhua) [inline inventory/allinone-ha.yaml:23]: Maybe named this in short like 'nodepool_builder_state'? and consider to rename for all of following vars.

File: playbooks/zookeeper.yaml [hunk 3]
```diff
@@ -38,9 +38,4 @@
       with_items: "{{ groups['zookeeper'] }}"
 
     - name: Restart zookeeper services
-      service: name=zookeeper state=restarted
-
-  post_tasks:
-    - name: Run zookeeper validation
-      include_role:
-        name: test.zookeeper
+      service: name=zookeeper state=restarted
```
Comment (huangtianhua) [inline playbooks/zookeeper.yaml]: no new line here, is it ok?

File: playbooks/zuul-fingergw.yaml [hunk 4]
```diff
@@ -42,10 +40,3 @@
               - rotate 7
               - daily
               - notifempty
-
-  post_tasks:
-    - name: Run zuul-fingergw validation.
-      include_role:
-        name: test.zuul-fingergw
-      vars:
-        zuul_service_zuul_fingergw_state: "{{ service_state|default('started') }}"
```
Comment (huangtianhua) [inline playbooks/zuul-fingergw.yaml:51]: same

File: playbooks/zuul-merger.yaml [hunk 5]
```diff
@@ -41,10 +39,3 @@
               - rotate 7
               - daily
               - notifempty
-
-  post_tasks:
-    - name: Run zuul-merger validation.
-      include_role:
-        name: test.zuul-merger
-      vars:
-        zuul_service_zuul_merger_state: "{{ service_state|default('started') }}"
```
Comment (huangtianhua) [inline playbooks/zuul-merger.yaml:50]: same

File: inventory/test-allinone-ha.yaml (hunk 1 above)
Comment (liusheng) [reply inventory/test-allinone-ha.yaml:87]: this is used by unison, not we do not need this when using rsync to sync logs.

File: playbooks/nodepool-builder.yaml [hunk 6]
```diff
@@ -67,11 +67,9 @@
       vars:
         nodepool_task_manager:
           - service
-        # disable nodepool launcher service
-        nodepool_file_nodepool_launcher_service_manage: false
-        nodepool_file_nodepool_launcher_service_config_manage: false
-        nodepool_service_nodepool_launcher_manage: false
-        nodepool_service_nodepool_builder_state: "{{ service_state|default('started') }}"
```
Comment (huangtianhua) [inline playbooks/nodepool-builder.yaml:74]: Here we delete the var, then where we use it?

File: inventory/allinone-ha.yaml (hunk 2 above)
Comment (liusheng) [reply inventory/allinone-ha.yaml:23]: this var name is defined in the role `https://git.openstack.org/openstack/ansible-role-nodepool`, here we just override this variable value to `stopped`

File: playbooks/zookeeper.yaml (hunk 3 above)
Comment (liusheng) [reply playbooks/zookeeper.yaml]: ok, will add

File: playbooks/zuul-fingergw.yaml (hunk 4 above)
Comment (liusheng) [reply playbooks/zuul-fingergw.yaml:51]: ok

File: playbooks/zuul-merger.yaml (hunk 5 above)
Comment (liusheng) [reply playbooks/zuul-merger.yaml:50]: ok

File: playbooks/nodepool-builder.yaml (hunk 6 above)
Comment (liusheng) [reply playbooks/nodepool-builder.yaml:74]: these variables have default values in the role `ansible-role-nodepool`, and as default, the launcher and builder service are enabled and started state. previously we do not start builder when deploy launcher and do not start launcher when deploy builder. now we have set these value as false in `inventory/group_vars/nodepool.yaml` and enable launcher when deploy launcher, enable builder when deploy builder.

Comment (kiwik) [reply playbooks/nodepool-builder.yaml:74]: @liu-sheng thank you for this clarification.

**Review by kiwik — APPROVED** [review]
LGTM

File: inventory/allinone-ha.yaml (hunk 2 above)
Comment (huangtianhua) [reply inventory/allinone-ha.yaml:23]: ok, thanks.

**Review by huangtianhua — APPROVED** [review]
LGTM

**Pull request merged**